docker_images/sklearn/tests/test_api_tabular_classification.py

```diff
@@ -67,26 +76,90 @@ def test_success_code(self):
         assert type(content) == list
         assert len(content) == expected_output_len
 
-    def test_malformed_input(self):
+    def test_wrong_sklearn_version_warning(self):
+        # if the wrong sklearn version is used the model will be loaded and
+        # gives an output, but warnings are raised. This test makes sure the
+        # right warnings are raised and that the output is included in the
+        # error message.
+        self._check_requirement(self.case_data["old_sklearn"] and self._can_load())
+
+        data = self.data
         with TestClient(self.app) as client:
-            response = client.post("/", data=b"Where do I live ?")
+            response = client.post("/", json={"inputs": data})
 
-        self.assertEqual(
-            response.status_code,
-            400,
-        )
+        assert response.status_code == 400
         content = json.loads(response.content)
-        self.assertEqual(set(content.keys()), {"error"})
+        assert "error" in content
+        assert "warnings" in content
+        wrong_versions = [
+            w for w in content["warnings"] if "Trying to unpickle estimator" in w
+        ]
+        assert len(wrong_versions)
+        error_message = json.loads(content["error"])
+        assert error_message["output"] == self.expected_output
+
+    def test_cannot_load_model(self):
+        # test the error message when the model cannot be loaded on a wrong
+        # sklearn version
+        self._check_requirement(not self.case_data["loads_on_new_sklearn"])
 
-    def test_missing_columns(self):
-        data = self.data["data"].copy()
-        data.pop(next(iter(data.keys())))
+        data = self.data
+        with TestClient(self.app) as client:
+            response = client.post("/", json={"inputs": data})
+
+        assert response.status_code == 400
+        content = json.loads(response.content)
+        assert "error" in content
+        assert "An error occurred while loading the model:" in content["error"]
+
+    @parameterized.expand(
+        [
+            ("add", "The following columns were given but not expected:"),
+            ("drop", "The following columns were expected but not given:"),
+        ]
```

Comment:
There could also be a test case for both add and drop.

Reply:
Adding it added complexity to the test, don't love it, but it's there now.